src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -1646,9 +1648,12 @@ private static NamedTypeSymbol GetNonDisplayClassContainer(NamedTypeSymbol type)
                 {
                     if (IsViableSourceMethod(candidateMethod, desiredMethodName, desiredTypeParameters, sourceMethodMustBeInstance))
                     {
+                        MethodSymbol sourceMethod = new EECompilationContextMethod(candidateSubstitutedSourceMethod.DeclaringCompilation!, candidateMethod.OriginalDefinition);
```

Comment:
>! [](http://example.com/codeflow?start=136&length=1)

Is `!` necessary?

Reply:
> Is ! necessary?

Yes, despite the assert above, a nullable warning is reported here. I will open an issue after the change is merged.